File: src/server/bo/ProjectBO.py
from .BusinessObject import BusinessObject

"""
Klasse Projekt

"""


class Project(BusinessObject):
    def __init__(self):
        super().__init__()

        self._projektname = ""
        self._laufzeit = None
        self._auftraggeber = ""
        self._availablehours = None

    def set_projektname(self, projektname):
        """Setzen des Projektnamen"""
        self._projektname = projektname

    def get_projektname(self):
        """Auslesen des Projektnamen"""
        return self._projektname

    def set_laufzeit(self, laufzeit):
        """Setzen der Projektlaufzeit"""
        self._laufzeit = laufzeit

    def get_laufzeit(self):
        """Auslesen der Projektlaufzeit"""
        return self._laufzeit

    def set_auftraggeber(self, auftraggeber):
        """Setzen des Auftraggebers"""
        self._auftraggeber = auftraggeber

    def get_auftraggeber(self):
        """Auslesen des Auftraggebers"""
        return self._auftraggeber

    def set_availablehours(self, availablehours):
        """Setzen der verfügbaren Stunden"""
        self._availablehours = availablehours

    def get_availablehours(self):
        """Auslesen der verfügbaren Stunden"""
        return self._availablehours

    @staticmethod
    def from_dict(dictionary=dict()):
        """Umwandeln eines Python dict() in ein Project()."""
        obj = Project()
        obj.set_projektname(dictionary["projektname"])
        obj.set_laufzeit(dictionary["laufzeit"])
        obj.set_auftraggeber(dictionary["auftraggeber"])
        obj.set_availablehours(dictionary["availablehours"])
        return obj

File: frontend/src/api/ProjectBO.js
import BusinessObject from "./BusinessObject";

/**
 * Represents a Project
 */
export default class ProjectBO extends BusinessObject {
  constructor(aProjektname, aLaufzeit, aAuftraggeber, aAvailablehours) {
    super();
    this.projektname = aProjektname;
    this.laufzeit = aLaufzeit;
    this.auftraggeber = aAuftraggeber;
    this.availablehours = aAvailablehours;
  }

  /**
   * Sets a new Projektname.
   * @param {String} aProjektname - the new projektname of the ProjektBO
   */
  setProjektname(aProjektname) {
    this.projektname = aProjektname;
    
  }

  /**
   * Gets the projektname.
   */
  getProjektname() {
    return this.projektname;
  }

    /**
   * Laufzeit Getter und Setter.
   */
  setLaufzeit(aLaufzeit) {
    this.laufzeit = aLaufzeit;
  }

  getLaufzeit() {
    return this.laufzeit;
  }
  /**
   * Auftraggeber Getter und Setter.
   */
  setAuftraggeber(aAuftraggeber) {
    this.auftraggeber = aAuftraggeber;
  }

  getAuftraggeber() {
    return this.auftraggeber;
  }
   /**
   * Availablehours Getter und Setter.
   */
  setAvailablehours(aAvailablehours) {
    this.availablehours = aAvailablehours;
  }

  getAvailablehours() {
    return this.availablehours;
  }

  static fromJSON(projects) {
    let result = [];

    if (Array.isArray(projects)) {
      projects.forEach((p) => {
        Object.setPrototypeOf(p, ProjectBO.prototype);
        result.push(p);
      });
    } else {
      // Es handelt sich offenbar um ein singuläres Objekt
      let p = projects;
      Object.setPrototypeOf(p, ProjectBO.prototype);
      result.push(p);
    }

    return result;
  }
}
